src/Microsoft.Graph.Core/Requests/RedirectHandler.cs

```diff
@@ -0,0 +1,172 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Text;
+    using System.Net.Http.Headers;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Net;
+    
+
+    /// <summary>
+    /// An <see cref="DelegatingHandler"/> implementation using standard .NET libraries.
+    /// </summary>
+    public class RedirectHandler : DelegatingHandler
+    {
+
+        private const int maxRedirects = 5;
```

Comment:
@MIchaelMainer  One requirement that I received from community members was the ability to configure that redirect value per request.  The idea is that we attach some kind of policy object to the request and then the RedirectHandler grabs the policy object from the request and calls into it to determine whether to retry.  e.g.

      ((Policy)request.Properties["policy"]).ShouldRetry(response)

This would give the ability for calls to certain endpoints to either do more or less retries and make decisions based on context.  I don't think this is immediately necessary, but that was how I was planning to give the configurabilty in the future.